uint16 id
float64 u
float64 v
float64 x
float64 y
float64 z